### ISSUE TITLE
Code completion does not work - Apex Language Server shows Java memory error

### ISSUE THREAD

**Issue by chulai** [forcedotcom/salesforcedx-vscode#1101]
### Summary

Created SFDX Project on existing metadata folder and tried to use Apex Code Completion but it did not work. Restarting Vs Code does not help.

### Steps To Reproduce:

1.  Open any Apex Class.
1.  Find any reference to a method or class in the code.
1.  Right click on the reference and select 'Go To Definition'.

### Expected result

Vs Code should follow the reference and jump to the method within the class or the method in a separate class.

### Actual result

Nothing happens. However, Apex Language Server Output windows shows a Java memory error:

14:47:45.292 [Apex Prelude Service STARTING] INFO  a.j.l.i.s.ApexPreludeManagedService - Scanning built-in and system Apex types.
Feb 23, 2019 2:47:45 PM apex.jorje.parser.impl.BaseApexLexer dedupe
INFO: Deduped array ApexLexer.DFA22_transition. Found 7272051 shorts which is 13MB not including array overhead. Removed 6615472 shorts which is 12MB not counting array overhead. Took 14 ms.
14:47:51.302 [Apex Prelude Service STARTING] INFO  a.j.l.i.s.ApexPreludeManagedService - Patching standard library for System.Database.* methods
14:47:51.319 [Apex Prelude Service STARTING] INFO  a.j.l.i.s.ApexPreludeManagedService - Done patching standard library for System.Database.* methods
14:47:54.335 [Apex Prelude Service STARTING] INFO  a.j.l.i.s.ApexPreludeManagedService - Scanning built-in and system Apex types took 9045 ms.
14:47:54.356 [ApexIndexer STARTING] INFO  a.jorje.lsp.impl.index.ApexIndexer - Scanning user-defined types.
Feb 23, 2019 2:48:23 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:192)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:99)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.lang.OutOfMemoryError: Java heap space
	at org.eclipse.jdt.internal.core.nd.db.ShortString.getChars(ShortString.java:69)
	at org.eclipse.jdt.internal.core.nd.db.ShortString.compare(ShortString.java:155)
	at org.eclipse.jdt.internal.core.nd.field.FieldSearchIndex$SearchCriteriaToBtreeVisitorAdapter.compare(FieldSearchIndex.java:132)
	at org.eclipse.jdt.internal.core.nd.db.BTree.accept(BTree.java:628)
	at org.eclipse.jdt.internal.core.nd.db.BTree.accept(BTree.java:598)
	at org.eclipse.jdt.internal.core.nd.field.FieldSearchIndex.findBest(FieldSearchIndex.java:232)
	at apex.jorje.lsp.impl.index.ApexIndex.findExactTypeId(ApexIndex.java:202)
	at apex.jorje.lsp.impl.symbols.ApexSymbolProvider.getTypeIdDependentOnNamespace(ApexSymbolProvider.java:161)
	at apex.jorje.lsp.impl.symbols.ApexSymbolProvider.find(ApexSymbolProvider.java:131)
	at apex.jorje.semantic.symbol.resolver.StandardSymbolResolver.find(StandardSymbolResolver.java:428)
	at apex.jorje.semantic.symbol.resolver.rules.TwoPartTypeNameResolveRules$InnerTypeInSameNamespace.lookup(TwoPartTypeNameResolveRules.java:280)
	at apex.jorje.semantic.symbol.resolver.rules.TypeNameResolutionOrderUtil.getTypeFromTwoPartRules(TypeNameResolutionOrderUtil.java:130)
	at apex.jorje.semantic.symbol.resolver.rules.TypeNameResolutionOrderUtil.bindToType(TypeNameResolutionOrderUtil.java:62)
	at apex.jorje.semantic.symbol.resolver.StandardSymbolResolver.lookupTypeInfo(StandardSymbolResolver.java:374)
	at apex.jorje.semantic.symbol.resolver.StandardSymbolResolver.lookupTypeInfoIdentifiers(StandardSymbolResolver.java:180)
	at apex.jorje.semantic.symbol.resolver.TypeRefResolvers$ClassTypeResolver.resolve(TypeRefResolvers.java:148)
	at apex.jorje.semantic.symbol.resolver.StandardSymbolResolver.lookupTypeInfo(StandardSymbolResolver.java:142)
	at apex.jorje.semantic.ast.statement.VariableDeclarationStatements.resolveType(VariableDeclarationStatements.java:171)
	at apex.jorje.semantic.ast.visitor.PostTypeResolveVisitor.visit(PostTypeResolveVisitor.java:160)
	at apex.jorje.semantic.ast.visitor.PostTypeResolveVisitor.visit(PostTypeResolveVisitor.java:56)
	at apex.jorje.semantic.ast.statement.VariableDeclarationStatements.traverse(VariableDeclarationStatements.java:84)
	at apex.jorje.semantic.ast.statement.BlockStatement.traverse(BlockStatement.java:140)
	at apex.jorje.semantic.ast.statement.MethodBlockStatement.traverse(MethodBlockStatement.java:75)
	at apex.jorje.semantic.ast.member.Method.traverse(Method.java:104)
	at apex.jorje.semantic.ast.compilation.UserClass.traverse(UserClass.java:129)
	at apex.jorje.semantic.compiler.CodeUnit.traverse(CodeUnit.java:193)
	at apex.jorje.semantic.compiler.CompilerOperations$PostTypeResolveOperation.invoke(CompilerOperations.java:84)
	at apex.jorje.semantic.compiler.CompilerStageProcessor.processFirstStage(CompilerStageProcessor.java:134)
	at apex.jorje.semantic.compiler.CompilerStageProcessor.processToStage(CompilerStageProcessor.java:103)
	at apex.jorje.semantic.compiler.ApexCompiler.compile(ApexCompiler.java:120)
	at apex.jorje.lsp.impl.services.StandardCompilerService.compileByPassValidation(StandardCompilerService.java:153)
	at apex.jorje.lsp.impl.tests.StandardTestService.lambda$getTestMethods$0(StandardTestService.java:88)

[Info  - 2:49:29 PM] Connection to server got closed. Server will restart.
[Error - 2:49:29 PM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.listen.readable.on (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close [as _onclose] (net.js:596:12)
[Error - 2:49:29 PM] Request textDocument/codeLens failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.listen.readable.on (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close [as _onclose] (net.js:596:12)
[Error - 2:49:29 PM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.listen.readable.on (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close [as _onclose] (net.js:596:12)
[Error - 2:49:29 PM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.listen.readable.on (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close [as _onclose] (net.js:596:12)
[Error - 2:49:29 PM] Request textDocument/definition failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.listen.readable.on (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close [as _onclose] (net.js:596:12)
[Error - 2:49:29 PM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.listen.readable.on (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close [as _onclose] (net.js:596:12)
[Error - 2:49:29 PM] Request textDocument/definition failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.listen.readable.on (C:\Users\myuser\.vscode\extensions\salesforce.salesforcedx-vscode-apex-45.3.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at Socket.emit (events.js:187:15)
	at Pipe.Socket._destroy._handle.close [as _onclose] (net.js:596:12)
14:49:31.066 [Apex Prelude Service STARTING] INFO  a.j.l.i.s.ApexPreludeManagedService - Scanning built-in and system Apex types.
Feb 23, 2019 2:49:31 PM apex.jorje.parser.impl.BaseApexLexer dedupe
INFO: Deduped array ApexLexer.DFA22_transition. Found 7272051 shorts which is 13MB not including array overhead. Removed 6615472 shorts which is 12MB not counting array overhead. Took 13 ms.
14:49:33.507 [ApexIndexer STARTING] INFO  a.jorje.lsp.impl.index.ApexIndexer - Scanning user-defined types.
14:49:33.506 [Apex Prelude Service STARTING] INFO  a.j.l.i.s.ApexPreludeManagedService - Scanning built-in and system Apex types took 2442 ms.

### Additional information

I have tried with both, Java 8 32-bit and 64-bit versions. Also, added explicit setting to point to the location where I installed Java 8: "salesforcedx-vscode-apex.java.home": "C:\\Program Files (x86)\\Java\\jdk1.8.0_191\\"

**VS Code Version**: 1.3.1

**SFDX CLI Version**: sfdx-cli/6.54.4-c01dcc69db (windows-x86) node-v8.9.4

**Java version**: 1.8.0_191-b12

**OS and Version**: Windows 10, 64-bit

**Comment by praksb** [comment]
@chulai It looks like the Apex language server crashes while indexing user-defined types. This is why you don't see references or go-to-defintion working. Can you share the number of classes and triggers in your project? Looking at the log line 

14:49:33.507 [ApexIndexer STARTING] INFO a.jorje.lsp.impl.index.ApexIndexer - Scanning user-defined types.

it also looks like your server restarted, did it complete scanning user-defined types or did it crash again?

**Comment by chulai** [comment]
@praksb classes: 3670. triggers: 108. I'm not counting the -meta.xml files, btw. It's a large project. However, I don't have this problem on another computer with the same project. It just takes a little time first time to navigate the reference but I don't get any error as in the first computer.

I'm not sure if it finished scanning all user-defined types. I do remember I left Vs Code to scan the files for a while in a couple of times with did not change the outcome.

Regarding ALS restarting, I don't think so. I remember that last lines came up only after I tried to browser another reference.

Is there a setting somewhere where I could increase the JVM memory if that could help? 

Worth noting that if I create a brand new SFDX project with a couple of Apex Classes the Code Completion features work just fine in the offending computer. So the size of the project seems to play a role in this issue.

Thanks!

**Comment by praksb** [comment]
@chulai From the call stack in the log you provided the issue you are seeing here is related to: 
https://github.com/forcedotcom/salesforcedx-vscode/issues/1103

When the test panel is open we try to scan all the classes in your project to get the complete list of test classes and test methods. Given that the number of test classes and test methods are pretty high in your case we are running out of memory trying to compile and get the results. I am going to close this issue out as a duplicate of: https://github.com/forcedotcom/salesforcedx-vscode/issues/1103 and use the other issue for tracking it at our end.

**Comment by praksb** [comment]
Duplicate of https://github.com/forcedotcom/salesforcedx-vscode/issues/1103

**Comment by praksb** [comment]
@chulai I have responded to your comment on #1103 as well. The StandardTestService that is throwing the out of memory exception is ran at startup also (not just when test panel is open as I previously commented) and compiles all the classes (not just test classes). ApexIndexer does this as well during startup which is causing two full compilations (of all classes) which ends up with an out of memory exception. The fix that we are working on would address both since StandardTestService would no longer be kicking off new compilations and would just consume the results of the ApexIndexer.

As for increasing the JVM memory size, I believe the only way you could do that is when the Apex LSP java process is started but our extensions currently do not make this configurable by the user.

**Comment by chulai** [comment]
@praksb thanks for the clarification. I will wait for the fix then.

**Comment by ClemABT** [comment]
Having the same issue since 2 days on a large org using version 45.12.1

**Comment by praksb** [comment]
@ClemABT Can you please open a separate issue? Please include your machine hardware configuration (CPU, RAM), Operating system as well as info on your org like number of classes, triggers. Also, include the call stack.

**Comment by OneMadhu** [comment]
I faced same issue, I have around 300+ apex and 250+ aura components.  I got same error as mentioned above. 

Below steps helped to resolve issue. 

-  Deleted unnecessary files and brought down files count to 100. 
-  Deleted apex.db file from \.sfdx\tools  and restarted VS.

Code completion works.
